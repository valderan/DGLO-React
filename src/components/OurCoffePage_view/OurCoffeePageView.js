import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../Header';
import Footer from '../Footer';
import Item from '../Item';

import './coffeepage.sass';

export default class OurCoffeePageView extends React.Component {


    render () {

        const {items, isLoad, countryFilter, searchFilter} = this.props;

        return (        
            <>
                <ContentPage>

                    <Filter countryFilter={countryFilter} searchFilter={searchFilter}/>

                    <ProductsList> 
                        
                        {
                            items.map( (item, index) => {
                                return(
                                    <WrapLink key={index} isLoad={isLoad} item={item}/>
                                )
                            })
                        }

                    </ProductsList>

                </ContentPage>
            </>
        )
    }

}


function Filter({countryFilter, searchFilter}) {
    return (
    <div className="row">
        <div className="col-lg-4 offset-2">
            <form action="#" className="shop__search">
                <label className="shop__search-label" htmlFor="filter">Looking for</label>
                <input onChange={ (e) => { searchFilter(e.target.value) }} 
                        onClick={ (e) => { searchFilter(e.target.value) }}
                        id="filter" type="text" 
                        placeholder="start typing here..." 
                        className="shop__search-input"/>
            </form>
        </div>
        <div className="col-lg-4">
            <div className="shop__filter">
                <div className="shop__filter-label">
                    Or filter
                </div>
                <div className="shop__filter-group">
                    <button onClick={() => countryFilter('Brazil')} className="shop__filter-btn">Brazil</button>
                    <button onClick={() => countryFilter('Kenya')} className="shop__filter-btn">Kenya</button>
                    <button onClick={() => countryFilter('Columbia')} className="shop__filter-btn">Columbia</button>
                </div>
            </div>
        </div>
    </div>
    )
}

function WrapLink({isLoad, item}) {
    const url = `/product/${item.name}`

    if (!isLoad) {
        return (
            <Link to={url} className="nolink">
                <Item item={item} isLoad={false} />
            </Link>
        )

    } else {

        return <Item item={item} isLoad={true} />
    }
}

function ProductsList({children}) {
    return (
        <div className="row">
            <div className="col-lg-10 offset-lg-1">
                <div className="shop__wrapper">
                    {children}
                </div>
            </div>
        </div>
    )
}

function ContentPage({children}) {
    return (
        <>
            <div className="banner">
                <div className="container">
                    <div className="row">
                        <div className="col-lg-6">
                             <Header />
                        </div>
                    </div>
                    <h1 className="title-big">Our Coffee</h1>
                </div>
            </div>

            <section className="shop">
            <div className="container">
                <div className="row">
                    <div className="col-lg-4 offset-2">
                        <img className="shop__girl" src="img/coffee_girl.jpg" alt="girl"/>
                    </div>
                    <div className="col-lg-4">
                        <div className="title">About our beans</div>
                        <img className="beanslogo" src="logo/Beans_logo_dark.svg" alt="Beans logo"/>
                        <div className="shop__text">
                            Extremity sweetness difficult behaviour he of. On disposal of as landlord horrible.
                            <br/><br/>
                            Afraid at highly months do things on at. Situation recommend objection do intention<br/>
                            so questions. <br/>
                            As greatly removed calling pleased improve an. Last ask him cold feel<br/>
                            met spot shy want. Children me laughing we prospect answered followed. At it went<br/>
                            is song that held help face.
                        </div>
                    </div>
                </div>
                <div className="line"></div>
                <div className="bottom30"></div>
                {children}
                
            </div>
            </section>
            <Footer />
        </>
    )
}