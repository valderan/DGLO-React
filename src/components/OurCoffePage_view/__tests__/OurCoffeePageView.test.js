import React from 'react';
import {shallow} from 'enzyme';

import OurCoffeePageView from '../OurCoffeePageView';

describe('Testing <OurCoffeePageView />', () => {
  
  it('urCoffeePageView rendered correctly', () => {
    const coffePage = shallow(<OurCoffeePageView items={[]}/>);
    expect(coffePage).toMatchSnapshot();
  });


  // items, isLoad, countryFilter, searchFilter, error, errorString
  it('Check prop [items]', () => {
    const props = {
      items:[], 
      isLoad: true,
      countryFilter: ()=>{},
      searchFilter: ()=>{},
      error:false,
      errorString: '' 
    }

    const wrapper = shallow(<OurCoffeePageView {...props}/>);
    expect(wrapper.prop('items')).toBeArray();
  
  })

})