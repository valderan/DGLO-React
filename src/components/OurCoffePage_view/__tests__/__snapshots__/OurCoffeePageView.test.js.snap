// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <OurCoffeePageView /> urCoffeePageView rendered correctly 1`] = `
<Fragment>
  <ContentPage>
    <Filter
      countryFilter={[Function]}
    />
    <ProductsList>
      <WrapLink
        isLoad={false}
        item={
          Object {
            "country": "Brazil",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet.",
            "name": "Death Wish Bean",
            "price": "12.99$",
            "url": "https://images-na.ssl-images-amazon.com/images/I/91DlZZBjxzL._SX522_.jpg",
          }
        }
        key="0"
      />
      <WrapLink
        isLoad={false}
        item={
          Object {
            "country": "Brazil",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet.",
            "name": "AROMISTICO Coffee 1 kg",
            "price": "6.99$",
            "url": "https://images-na.ssl-images-amazon.com/images/I/71qBQnpQFYL._SL1500_.jpg",
          }
        }
        key="1"
      />
      <WrapLink
        isLoad={false}
        item={
          Object {
            "country": "Kenya",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet.",
            "name": "Solimo Coffee Beans 2 kg",
            "price": "10.73$",
            "url": "https://images-na.ssl-images-amazon.com/images/I/815O9ktyfUL._SL1500_.jpg",
          }
        }
        key="2"
      />
      <WrapLink
        isLoad={false}
        item={
          Object {
            "country": "Kenya",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet.",
            "name": "Black Rifle Coffee",
            "price": "19.75$",
            "url": "https://images-na.ssl-images-amazon.com/images/I/91vwF6Kh8IL._SX522_.jpg",
          }
        }
        key="3"
      />
      <WrapLink
        isLoad={false}
        item={
          Object {
            "country": "Columbia",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet.",
            "name": "Presto Coffee Beans 1 kg",
            "price": "15.99$",
            "url": "https://images-na.ssl-images-amazon.com/images/I/91Ryk2gKejL._SL1500_.jpg",
          }
        }
        key="4"
      />
      <WrapLink
        isLoad={false}
        item={
          Object {
            "country": "Columbia",
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Lorem ipsum dolor sit amet.",
            "name": "Organic Coffee One Cup",
            "price": "24.99$",
            "url": "https://images-na.ssl-images-amazon.com/images/I/51M2fmEXTIL.jpg",
          }
        }
        key="5"
      />
    </ProductsList>
  </ContentPage>
</Fragment>
`;
